relay-event-schema/src/protocol/replay.rs

```diff
@@ -299,4 +299,35 @@ mod tests {
         assert_eq!(event, Annotated::from_json(input).unwrap());
         assert_eq!(output, event.to_json().unwrap());
     }
+
+    #[test]
+    fn test_length_validation() {
+        // String values do not pass validation due to length.
+
+        let input = format!(r#"{{"dist":"{}"}}"#, str::repeat("a", 65));
+        let event = Annotated::<Replay>::from_json(&input);
+        assert!(event.is_err());
```

Comment:
This validates the replay deserialization, and content validation is done in normalization steps. Light normalization should catch the lengths, and the test would look something like the following:

```rs
let json = r#"..."#;
let event = Annotated::<Replay>::from_json(json).unwrap();
let config = LightNormalizationConfig {
  enable_trimming: true,
  ..Default::default()
};
light_normalize_event(&event, config).unwrap();
// TODO: ensure the field has an error
```
You may want to move these tests to `relay-event-normalization` where normalization functionality is defined.

Reply:
Added a test on https://github.com/getsentry/relay/pull/2587/files#diff-c3b1b3a23c63306f4043665940e399a231d91ffc3795a95a45371ca940893a58R381